packages/devtools/test/settings_controller_test.dart

```diff
@@ -0,0 +1,45 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+@TestOn('vm')
+import 'dart:async';
+import 'dart:io';
+
+import 'package:devtools/src/settings/settings_controller.dart';
+import 'package:test/test.dart';
+import 'package:vm_service/vm_service.dart';
+
+import 'support/flutter_test_driver.dart';
+import 'support/flutter_test_environment.dart';
+
+void main() async {
+  SettingsController settingsController;
+  final env = FlutterTestEnvironment(
+    const FlutterRunConfiguration(withDebugger: true),
+  );
+  group('settings controller test', () {
+    test('entering', () async {
+      await env.setupEnvironment();
+
+      bool isOnFlutter = false;
+      List<Flag> flags = [];
+      settingsController =
+          SettingsController(onIsAnyFlutterAppReady: (isAnyFlutterApp) {
+        isOnFlutter = isAnyFlutterApp;
+      }, onFlagListReady: (flagList) {
+        flags = flagList.flags;
+      });
+      expect(isOnFlutter, false);
+      expect(flags, []);
+      await settingsController.entering();
+      expect(isOnFlutter, true);
+
+      final flagList = await env.service.getFlagList();
```

Comment:
also add a test that some arbitrarily chosen existing flag that should be set is set.
Otherwise if env.service.getFlagList() got broken, this test would continue to succeed.

Reply:
is one flag enough to cover this? Sorry to ask this but how can I know which flag should be set?

Reply:
Two flags is probably enough. I definitely wouldn't bother with more than 3 flags. As a good rule of thumb unittesting is checking for 3 values is always plenty and sometimes two is enough.
Here I would think two is plenty and my main concern would be just avoiding flags that might go away causing a spurious test failure. These two are probably fine and are unlikely to have description text that will change.
<img width="118" alt="Screen Shot 2019-08-20 at 1 31 53 PM" src="https://user-images.githubusercontent.com/1226812/63382203-172a6d80-c34f-11e9-8f00-f96092b4e349.png">
<img width="158" alt="Screen Shot 2019-08-20 at 1 32 07 PM" src="https://user-images.githubusercontent.com/1226812/63382211-198cc780-c34f-11e9-8b2a-e267847a1191.png">

Reply:
Ok, changed as you suggested! Thanks for the rule of thumb and explanation 👍 :D